agagd/agagd_core/views/players_profile.py

```diff
@@ -45,56 +45,68 @@ def get(self, request, *args, **kwargs):
             Q(pin_player__exact=player_id)
         ).values("pin_player", "rating", "sigma")
 
-        # compute additional tables for opponents & tournament info. here
-        # TODO: refactor this into something nicer.
-        opponent_data = {}
-        tourney_data = {}
-        for game in player_games:
-            try:
-                t_dat = tourney_data.get(game.tournament_code.pk, {})
-                t_dat["tournament"] = game.tournament_code
-                t_dat["won"] = t_dat.get("won", 0)
-                t_dat["lost"] = t_dat.get("lost", 0)
-
-                # Set default game_date to None
-                game_date = None
-
-                # Check for 0000-00-00 dates
-                if game.game_date != u"0000-00-00":
-                    game_date = game.game_date
-
-                t_dat["date"] = t_dat.get("date", game_date)
-
-                op = game.player_other_than(player)
-                opp_dat = opponent_data.get(op, {})
-                opp_dat["opponent"] = op
-                opp_dat["total"] = opp_dat.get("total", 0) + 1
-                opp_dat["won"] = opp_dat.get("won", 0)
-                opp_dat["lost"] = opp_dat.get("lost", 0)
-                if game.won_by(player):
-                    opp_dat["won"] += 1
-                    t_dat["won"] += 1
-                else:
-                    opp_dat["lost"] += 1
-                    t_dat["lost"] += 1
-                opponent_data[op] = opp_dat
-                tourney_data[game.tournament_code.pk] = t_dat
-            except ObjectDoesNotExist:
-                print("failing game_id: %s" % game.pk)
-
-        opp_table = PlayersOpponentTable(opponent_data.values())
-        RequestConfig(request, paginate={"per_page": 10}).configure(opp_table)
-
-        t_table = PlayersTournamentTable(
-            tourney_data.values(),
-            sorted(
-                tourney_data.values(),
-                key=lambda d: d.get("date", date.today()) or date.today(),
-                reverse=True,
-            ),
```

Comment:
The purpose of giving `tourney_data` twice to the  `PlayersTournamentTable` is unclear to me.  Especially since the first time is unsorted while the second time is sorted.  This may represent some kind of logic which is not reflected in the new implementation.